src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -441,7 +441,12 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
             // We won't track and only stamp function specific details on the RequestTelemetry
             // created by SDK via Activity when function ends
             if (Activity.Current == null ||
-                DictionaryLoggerScope.GetMergedStateDictionary().ContainsKey("MS_IgnoreActivity"))
+                // Activity is tracked, but Functions wants to ignore it:
+                DictionaryLoggerScope.GetMergedStateDictionary().ContainsKey("MS_IgnoreActivity") ||
+                // Functions create another RequestTrackingTelemetryModule to make sure first request is tracked (as ASP.NET Core starts before web jobs)
+                // however at this point we may discover that RequestTrackingTelemetryModule is disabled by customer and even though Activity exists, request won't be tracked
+                // So, if we've got AspNetCore Activity and EnableHttpTriggerExtendedInfoCollection is false - track request here.
+                (!_loggerOptions.HttpAutoCollectionOptions.EnableHttpTriggerExtendedInfoCollection && Activity.Current?.OperationName == "Microsoft.AspNetCore.Hosting.HttpRequestIn"))
```

Comment:
Awesome! I wasn't sure how to differentiate the Activity between ServiceBus (and others) and Http -- but checking the OperationName is perfect.

I'm building this now and going to test in the private stamp. I was able to verify that it is, indeed, a bug -- we'd drop any cold start requests.